gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcher.java

```diff
@@ -73,48 +74,20 @@ public QueryServiceEntityFetcher(
   @Override
   public EntityFetcherResponse getEntities(
       EntitiesRequestContext requestContext, EntitiesRequest entitiesRequest) {
+    Map<String, AttributeMetadata> attributeMetadataMap =
+        attributeMetadataProvider.getAttributesMetadata(
+            requestContext, AttributeScope.valueOf(entitiesRequest.getEntityType()));
+    // Validate EntitiesRequest
+    entitiesRequestValidator.validate(entitiesRequest, attributeMetadataMap);
```

Comment:
Can this help you remove some if checks in the code below given that validations will catch the issues?

Reply:
Which validation below that you're referring to? I can't find additional validation after this line, mostly extraction.

Reply:
Just wondering if it can simplify some code down below.

Reply:
Ah, I see. Mostly is transforming the data. It can be extracted in the next re-factoring.